### ISSUE TITLE
API change for conf_interval()

### ISSUE THREAD

**Issue by tritemio** [lmfit/lmfit-py#263]
In lmfit 0.8.3 we could do:

```
fit_res = model.fit(y, x=x)
ci = lm.conf_interval(fit_res)
```

This raises an error in lmfit 0.9.1. I couldn't find any mention to a changed API for `conf_interval` in the docs.

We should either add a note to the docs or restore the old API.

**Comment by newville** [comment]
@tritemio   definitely needs a doc change.  Sorry I forgot to do this.  The API had to be changed to work with the new `MinimizerResult`.

**Comment by tritemio** [comment]
@newville no need to apologize. How do we call `lm.conf_interval` with results from a  model fit in lmfit 0.9?

**Comment by licode** [comment]
probably you can see https://github.com/lmfit/lmfit-py/issues/251

**Comment by tritemio** [comment]
@licode, my question here is how to compute `lm.conf_interval` when using `lm.Model` to fit, like in the first example in this issue.

**Comment by newville** [comment]
@tritemio  actually @licode  gave you a very good hint. 

```python
ci = lm.conf_interval(fit_res, fit_res)
```
is probably what you want.  I suppose in  that sense, we could allow the second argument to be `None` as long as the first argument had `chisqr` and `params` attributes.....   Or, well, probably adding a `conf_interval` method to `ModelFit` is simpler....

**Comment by tritemio** [comment]
@newville, ok thanks, it wasn't obvious to me without looking at the implementation. 

Anyway, if it's only an implementation detail I would prefer maintaining the old API, less broken scripts.

**Comment by newville** [comment]
@tritemio it's not quite an implementation detail that 

```python
ci = lm.conf_interval(fit_res, fit_res)
```
works.   The 1st arg of `conf_interval` needs to be something that subclasses `Minimizer`, so that `.leastsq()` can be run.  The 2nd arg needs to be something that mimics a `MinimizerResult` in that it has the updated `params` (with best values and uncertainties), and has a `chisqr`.     It's sort of accidental that `ModelResult` satisfies both these needs.    Then again, it was sort of accidental that 
`conf_interval(fit_res)` worked in 0.8.3.

As for scripts breaking:  0.9.0 changed the core API of minimization and broke many scripts. It's kind of a major feature of the release, and was advertised many times. The PR (#230) was made in May, and there were multiple pleas for comments and suggestions.  It was merged six weeks later to allow time for comments and feedback.  The release of this "master" as "0.9.0" waited another 3 months for comments.   Apparently no one cared to try this feature out in that time.    

Of course, it's never too late to complain.  But, the API for `conf_interval()` did change, and we're not going back.  The new API is actually documented, it's just not mentioned as part of the clear warning about the changed API.   

I'm -1 on making the second argument of `conf_interval` default to `None`.   We could add a `conf_interval` method to `ModelResult`, but that's a different API too.

**Comment by tritemio** [comment]
@newville no complain, just reporting. I just recently noticed this issue while running some old scripts that I didn't try before. Interestingly, all the other scripts I tested before the 0.9 release did not break because I'm mostly using the `Model` interface.

The minimizer API change had good justification and there was time for discussion, nothing to say about that.

My point is that even when the API changes, there is a bigger impact in changing two functions instead of only one. If, on a technical basis, there is a reason for `conf_interval` to don't support the old API then I'm ok with it.

If `conf_interval` must not support the old API, then adding a method to `ModelResult` is a good alternative, better than `lm.conf_interval(fit_res, fit_res)` that does not make much sense from an user point of view.
